File: OrganizationService/src/main/java/com/slavacom/organizationservice/entity/KanbanTaskPositions.java
package com.slavacom.organizationservice.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.util.UUID;

@Getter
@Setter
@Entity
@Table(name = "kanban_task_positions")
public class KanbanTaskPositions {
	@Id
	@GeneratedValue(strategy = GenerationType.UUID)
	@Column(name = "id", nullable = false)
	private UUID id;

	@Column(name = "project_id", nullable = false)
	private UUID projectId;

	@Column(name = "kanban_column_id", nullable = false)
	private UUID kanbanColumnId;

	@Column(name = "task_id", nullable = false)
	private UUID taskId; // ID задачи из Task Service

	@Column(name = "order_index", nullable = false)
	private Integer orderIndex; // порядок задач в колонке

	@UpdateTimestamp
	@Column(name = "updated_at")
	private Instant updatedAt;
}

File: OrganizationService/src/main/java/com/slavacom/organizationservice/entity/KanbanTaskPositions.kt
package com.slavacom.organizationservice.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.UpdateTimestamp
import java.time.Instant
import java.util.UUID

@Entity
@Table(name = "kanban_task_positions")
open class KanbanTaskPositions {
    @field:Id
    @field:GeneratedValue(strategy = GenerationType.UUID)
    @field:Column(name = "id", nullable = false)
    open var id: UUID? = null

    @field:Column(name = "project_id", nullable = false)
    open var projectId: UUID? = null

    @field:Column(name = "kanban_column_id", nullable = false)
    open var kanbanColumnId: UUID? = null

    @field:Column(name = "task_id", nullable = false)
    open var taskId: UUID? = null

    @field:Column(name = "order_index", nullable = false)
    open var orderIndex: Int? = null

    @field:UpdateTimestamp
    @field:Column(name = "updated_at")
    open var updatedAt: Instant? = null
}

